### PR TITLE
Add structure docs to new developer onboarding

### LINKED ISSUE #2218: Include high level architecture and overview of components in docs
When I was working on my first PR for the dashboard it took me a good amount of time to get a high level understanding of how the dashboard is structured. 

Even when I thought I understood a system built into the dashboard, I couldn't be 100% certain because there were many files and functions I had never closely evaluated.

In my limited open source experience it is common to not have documentation on the general architecture/code structure of a project, but it would have been helpful for me and might be helpful for others interested in contributing. 

Do we have some docs like these somewhere that I missed? If not, I am willing to write them, provided that people are willing to correct me where I go wrong.

### PR DESCRIPTION
**Pull request by cnwalker**
Looking for corrections, expansion ideas and any other feedback! closes #2218.

### CODE REVIEW

File: docs/devel/dashboard-structure.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+
+### Frontend Structure
+- The frontend makes calls to the api and renders received data. The frontend also transforms some data on the client and provides visualizations for the user. The frontend also makes calls to the api server to do things like exec into a container directly from the dashboard.
+
+- The frontend also automatically generates localized translations. You can generate translations manually by running `gulp generate-xtbs` or `gulp serve:prod`.
+
+- The frontend uses [angular](https://angular.io/), a javascript model-view-controller framework.
+
+### Tests
+- Tests can be found in the top level `/test/` directory. The test directory is structured to mirror the `/src/` directory.
```
Comment (maciaszczykm) [inline docs/devel/dashboard-structure.md]: Backend tests are inside `/src/app/backend` directory though.

File: docs/devel/code-structure.md [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+- Every api call hits `apihandler.go` which implements a series of handler functions to pass the results to resource-specific handlers.
+
+- The dashboard backend currently doesn't implement a cache, so calls to the dashboard api will always make fresh calls to the kubernetes apiserver.
+
+- The backend is written in [Golang](https://golang.org/).
+
+### Frontend
+- The frontend makes calls to the api and renders received data. The frontend also transforms some data on the client and provides visualizations for the user. The frontend also makes calls to the api server to do things like exec into a container directly from the dashboard.
+
+- The frontend also automatically generates localized translations. You can generate translations manually by running `gulp generate-xtbs` or `gulp serve:prod`.
```
Comment (maciaszczykm) [inline docs/devel/code-structure.md]: You can link other guides here.

File: docs/devel/code-structure.md [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+- Kube dashboard is split into a *backend api* and a *frontend app*. The api is located in the backend folder and the frontend is located in the frontend folder in the `src` directory.
+
+### Backend (API)
+- Every api call hits `apihandler.go` which implements a series of handler functions to pass the results to resource-specific handlers.
+
+- The dashboard backend currently doesn't implement a cache, so calls to the dashboard api will always make fresh calls to the kubernetes apiserver.
+
+- The backend is written in [Golang](https://golang.org/).
+
+### Frontend
```
Comment (maciaszczykm) [inline docs/devel/code-structure.md]: Mention Material design here.

**Comment by cnwalker** [comment]
Mentioned the localization guide and text conventions, ready for another look @maciaszczykm!

File: docs/devel/code-structure.md [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+- The frontend makes calls to the api and renders received data. The frontend also transforms some data on the client and provides visualizations for the user. The frontend also makes calls to the api server to do things like exec into a container directly from the dashboard.
+
+- The frontend also automatically generates localized translations. You can generate translations manually by running `gulp generate-xtbs` or `gulp serve:prod`. Take a look at the [localization guide](localization.md) and [text conventions](text-conventions.md) for more info.
+
+- The frontend uses [angular](https://angular.io/), a javascript model-view-controller framework along with [material design](https://material.angularjs.org/latest/getting-started) for cards, components, search bars, and all other visuals.
+
+- [Material design guidelines](https://material.io/guidelines/)
+
+- Production javascript is served via [google closure compiler](https://developers.google.com/closure/compiler/), so please make sure all functions and variables are properly type annotated.
```
Comment (floreks) [inline docs/devel/code-structure.md]: Change served to compiled or transpiled. It translates ES6 to ES3 and provides some optimization checks and features.

**Comment by cnwalker** [comment]
Updated and ready for another review @floreks!

**Comment by maciaszczykm** [comment]
No, LGTM. I am merging it.

**Pull request merged**